core/payment/src/models/payment.rs

```diff
@@ -115,6 +131,22 @@ impl ReadObj {
             details: base64::encode(&self.details),
         }
     }
+
+    pub fn into_signed_api_model(
+        self,
+        activity_payments: Vec<ActivityPayment>,
+        agreement_payments: Vec<AgreementPayment>,
+    ) -> api_model::Signed<api_model::Payment> {
+        api_model::Signed {
+            payload: self
+                .clone()
+                .into_api_model(activity_payments, agreement_payments),
+            signature: Some(Signature {
+                signature: self.signature.unwrap(),
+                signed_bytes: self.signed_bytes.unwrap(),
```

Comment:
You should never unwrap in production code, unless you checked conditions line before.